api/data.go

```diff
@@ -0,0 +1,69 @@
+package api
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+
+	"github.com/ONSdigital/dp-legacy-cache-api/models"
+	"github.com/ONSdigital/log.go/v2/log"
+)
+
+// GetDataSets reads all messages from the datastore
+func (api *API) GetDataSets(ctx context.Context) http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		log.Info(ctx, "calling get datasets handler")
+
+		results, err := api.dataStore.GetDataSets(ctx)
+		if err != nil {
+			log.Error(ctx, "error retrieving datasets from db", err)
+			return
+		}
+
+		// Setting the header and encoding the results to JSON
+		w.Header().Set("Content-Type", "application/json; charset=utf-8")
+		err = json.NewEncoder(w).Encode(results)
+		if err != nil {
+			log.Error(ctx, "error encoding results to JSON", err)
+			http.Error(w, "Internal Server Error", http.StatusInternalServerError)
+			return
+		}
+	}
+}
+
+// AddDataSets stores a message in the datastore
+func (api *API) AddDataSets(ctx context.Context) http.HandlerFunc {
```

Comment:
Is this a 'Data Set' or a 'Dataset' - if so the casing should be set appropriately.

Reply:
This is in many places so may need updating elsewhere if so.

Reply:
I think the casing can just be as a Data Set for now, these functions will be replaced with cache time handlers later on anyway.